### PR TITLE
update dependencies

### PR DESCRIPTION
**Pull request by bengerman13**
## Changes proposed in this pull request:

- update all the dependencies
- switch from unmaintained `pytest-watch` to maintained `pytest-watcher` (this fix is untested because of my current containerization setup)

*most* of these changes are around sqlalchemy 1.x to sqlalchemy 2.x, 

## Things to check

- For any logging statements, is there any chance that they could be logging sensitive data?
- Are log statements using a logging library with a logging level set? Setting a logging level means that log statements "below" that level will not be written to the output. For example, if the logging level is set to `INFO` and debugging statements are written with `log.debug` or similar, then they won't be written to the otput, which can prevent unintentional leaks of sensitive data.

## Security considerations

being up-to-date on dependencies reduces our exposure to known exploits. Getting sqlalchemy from 1.x to 2.x will also make it easier to stay up-to-date in the future

### CODE REVIEW

File: broker/api.py [hunk 1]
```diff
@@ -115,7 +115,7 @@ def last_operation(
         :rtype: LastOperation
         """
 
-        instance = ServiceInstance.query.get(instance_id)
+        instance = db.session.get(ServiceInstance, instance_id)
```
Comment (bengerman13) [inline broker/api.py:118]: this is a sqlalchemy 1.x to 2.x change - you'll see this pattern a ton. 
`ModelType.query.get(identifier)` is now `db.session.get(ModelType, identifier)`

File: broker/extensions.py [hunk 2]
```diff
@@ -4,5 +4,5 @@
 from broker.config import config_from_env
 
 config = config_from_env()
-db = SQLAlchemy()
+db = SQLAlchemy(disable_autonaming=True)
```
Comment (bengerman13) [inline broker/extensions.py:7]: this is a workaround for a new issue with `flask-sqlalchemy` - https://github.com/sqlalchemy/sqlalchemy/discussions/10461

File: broker/models.py [hunk 3]
```diff
@@ -1,6 +1,7 @@
 from enum import Enum
 from sqlalchemy.dialects import postgresql
 import sqlalchemy as sa
+from sqlalchemy.orm import mapped_column
```
Comment (bengerman13) [inline broker/models.py:4]: sqlalchemy 1.x -> 2.x change. 
`mapped_column` replaces `db.Column`, mostly as a drop-in

File: tests/integration/test_canceling_ops.py [hunk 4]
```diff
@@ -45,7 +45,7 @@
 
 
 @pytest.mark.parametrize("task_type", params)
-def test_noop_when_operation_canceled(task_type):
+def test_noop_when_operation_canceled(clean_db, task_type):
```
Comment (bengerman13) [inline tests/integration/test_canceling_ops.py:48]: I don't fully understand why this changed, but a bunch of tests couldn't find a valid session anymore, so shimming in the `clean_db` fixture gives them a session

File: pip-tools/requirements.in [hunk 5]
```diff
@@ -1,6 +1,9 @@
...
-sqlalchemy==1.4.22
+Flask-Migrate
+# updating flasksqlalchemy to 3+ hits this bug: https://github.com/pallets-eco/flask-sqlalchemy/issues/1268
+Flask-SQLAlchemy>3
+# flask-sqlalchemy<3 calls apis deprecated in flask 2.3
+flask
+sqlalchemy
 # pin acme/pyopenssl until https://github.com/certbot/certbot/issues/9828 is fixed
 acme<2
 pyopenssl<23.3.0
```
Comment (markdboyd) [inline pip-tools/requirements.in:9]: Should/can we remove some of these version constraints? I added them during my last attempt to update the package dependencies

Comment (bengerman13) [reply pip-tools/requirements.in:9]: I looked at the linked issue and it's still open, so I think we're still stuck on this one

**Review by markdboyd — APPROVED** [review]

**Pull request merged**